### ISSUE TITLE
Import S7 metadata into database

### ISSUE THREAD

**Issue by mkosmala** [SnapshotSerengetiScienceTeam/DataManagement#4]

**Comment by mkosmala** [comment]
Hi @palme516, I'm working on getting S7 and S8 into the database. But the cleaned metadata files for these seasons have slightly different formats. Particularly problematic, the S8 file omits seconds in the timestamp -- it only has minutes and hours. We'll need to rerun the script that produced this file. I hope you didn't edit it by hand. (Did you?) I can rerun the script, but I don't know which one it is. (@aliburchard, do you?)

**Comment by mkosmala** [comment]
Splitting this issue into an S7 one and an S8 one...

**Comment by mkosmala** [comment]
For S7, we have one capture that has two different invalid types (a la #43). We need to split this capture into two different ones. We can do that once the import into the database is complete, but I want a note here, so I don't forget to do it. The capture in question is S7, L10, roll 1, capture 3.

**Comment by mkosmala** [comment]
Another thing to note is that starting in S8, the photos are all properly renamed in the clean metadata file. (For example: S7_L10_R1_IMAG0001.JPG instead of just IMAG0001.JPG)

But for S7 and previous, it's not. I'll import the S7 metadata file as-is and then change image filenames for S1-S7 all in one go per #48.

**Comment by mkosmala** [comment]
Also, a question to address: should I also be importing S6_MORE? Or has that already been done. Will investigate.

**Comment by mkosmala** [comment]
Each new season metadata import will need to also have a 'season' file that describes that season. Format is one comma-separated line with fields:
season: integer
start date: date in format 'YYYY-MM-DD'
end date: date in format 'YYYY-MM-DD'
comments: quoted string of up to 500 characters

**Comment by mkosmala** [comment]
@palme516 @aliburchard We will need approximate start and end dates for season 7 and 8. Can either of you provide those?

**Comment by mkosmala** [comment]
We need another change to the cleaning scripts. Whatever was run on S7 and S8 didn't preserve the original JPG timestamp. We need both the old and new timestamps. Will make a separate issue for cleaning script changes.

**Comment by palme516** [comment]
I'm going to be uploading S9 to MSI shortly and will begin cleaning in the next few days (also S10 when it arrives in the mail from Michael Anderson). Could you please keep me up to date on which cleaning scripts are new vs. old and what changes might need to be made? Are there still cleaning script issues or did the primary issue result from viewing the data in Excel?

**Comment by mkosmala** [comment]
Please, please do not do anything with S9 until we have S7 and S8 sorted out. I have not touched any of the cleaning scripts. I have never used them and do not know what they contain. That's why I've been waiting for you. Please see #49 for what needs to be done.

**Comment by palme516** [comment]
Roger that. I was just going to upload S9 images to MSI while we worked through these issues. I'll work through S8 cleaning with the scripts we have been using this weekend and make sure when I meet with IT next week that our outputs from the new scripts are correct.

**Comment by mkosmala** [comment]
Oh, gosh, the new scripts. Ugh. Okay, here's a good plan: run the MSI scripts on S7 and S8 and make sure they produce EXACTLY THE SAME output as the current output. It's very possible that they'll create different numbers for their captures or images, and that would make for a **huge** mess. (It took Ali and I about 6 months to disentangle the mess that was S1-S6 because of exactly this sort of process.)

**Comment by palme516** [comment]
I am all for not creating huge messes. I won't run the new scripts on any new data until the check out exactly with what we've already got.

**Comment by mkosmala** [comment]
Okay... the metadata from the cleaned timestamp scripts from S7 have been imported into the database. 

In S7 there are:
486 total rolls
305,925 total capture events
304,671 capture events that went to Zooniverse (99.6% of total)
835,483 total images

Notes:
1. The roll times aren't right. I accidentally calculated roll times by site. So don't use this for CT effort yet until I fix them.
  TO DO: 
      (A) write script to fix S7 roll start and end dates in database
      (B) fix importation script to correctly calculate roll start and end dates
2. There was a bug in my script that I didn't catch in testing. It meant that 43 rolls were combined with one another in weird ways. I have since corrected the script and cleaned up the database. As a result, the index numbers in the database may not be in the order you expect them, and there are some big gaps in the numbering (due to deletions of entire rolls). This is no big deal for any sort of analysis or use of the database. Just wanted to note it here in case anyone was examining the database in detail and wondering what happened with the S7 database index numbering.

Once I fix the issue with the roll dates, I'll import S8 and then fix up the minor details in this thread and in #47, and then close them. Was hoping to finish this all today, but the bug I mention above cost me 4 hours....

**Comment by mkosmala** [comment]
I have fixed the roll start and stop times for S7. 

The script that uploads the cleaned season data no longer calculates roll start and stop times. Instead, there is a separate script that does that after the data are all imported. This is a much cleaner solution (and allows for easy re-calculation of start and stop times if some images are later invalidated). 

Now just need to address the small issues listed in this thread...

**Comment by mkosmala** [comment]
S7, L10, roll 1, capture 3 supposedly has two different Invalid codes. But the whole roll is Invalid 2, so I'm just going to leave it as-is.

**Comment by mkosmala** [comment]
S7 metadata is now all properly uploaded into the database.
